conda/core/link.py

```diff
@@ -186,14 +194,27 @@ def make_conda_meta_create_action(all_target_short_paths):
         python_entry_point_actions = ()
         pyc_compile_actions = ()
 
+    if package_info.repodata_record.preferred_env is not None:
+        preferred_env_prefix = preferred_env_to_prefix(package_info.repodata_record.preferred_env,
+                                                       context.root_prefix, context.envs_dirs)
+        application_entry_point_action = (make_application_entry_point_action(
+            preferred_env_prefix, package_info.repodata_record.name),)
+
+        private_envs_meta_action = tuple([CreatePrivateEnvMetaAction(transaction_context,
```

Comment:
private_envs_meta_action = CreatePrivateEnvMetaAction(transaction_context, package_info, target_prefix),

or

    private_envs_meta_action = (CreatePrivateEnvMetaAction(transaction_context, package_info, target_prefix),)

to split across lines

`tuple([])` makes a list object, then a tuple object.  It's just extra work.  Just use a tuple literal.